src/audio/dai-zephyr.c

```diff
@@ -306,24 +299,24 @@ dai_dma_cb(struct dai_data *dd, struct comp_dev *dev, uint32_t bytes,
 				}
 
 				if (sink_dev && sink_dev->state == COMP_STATE_ACTIVE)
-					ret = dma_buffer_copy_from_no_consume(dma_buf, sink_c,
-									      converter[j], bytes);
-err:
-				buffer_release(sink_c);
+					ret = dma_buffer_copy_from_no_consume(dd->dma_buffer,
+									      sink, converter[j],
+									      bytes);
 			}
 		}
+err:
```

Comment:
Looking at it again I think I made a mistake here, I moved the label out of the list loop.  It seems the loop does not stop after an error, and just continues on to the next iteration, as error label was there to just release the buffer. Will it jump correctly if it points to end of a loop block?